tests/test_targets.py

```diff
@@ -255,3 +260,43 @@ def test_custom_target_not_sent():
 
     assert target.value == b''
     assert target.multipart_filename is None
+
+
+class TestS3TargetUpload(TestCase):
```

Comment:
Let's use `pytest` style tests written as plain functions. The `boto3` client and `mock_s3` could be added as fixtures that the test function makes use of.

```python
def test_s3_upload(mock_s3, mock_client):
  ...
```